homeassistant/components/media_player/firetv.py

```diff
@@ -46,13 +55,20 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     host = config.get(CONF_HOST)
     port = config.get(CONF_PORT)
     device_id = config.get(CONF_DEVICE)
+    apps = config.get(CONF_APPS)
 
     try:
         response = requests.get(DEVICE_LIST_URL.format(host, port)).json()
         if device_id in response[CONF_DEVICES].keys():
-            add_devices([FireTVDevice(host, port, device_id, name)])
+            device = FireTVDevice(host, port, device_id, name)
+            add_devices([device])
             _LOGGER.info('Device %s accessible and ready for control',
                          device_id)
+
+            _LOGGER.debug("APPS: %s %s", (type(apps), apps))
+            for app, config in apps.items():
+                add_devices([FireTVApp(device, app, config)])
```

Comment:
Apps are better exposed using current_source / sources. They should not be standalone media players.